main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -399,11 +399,12 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
               val annVal = traverse(ann)(visitAnnotation(_, env0, uenv0, tenv))
               val tconstr = NamedAst.TypeConstraint(Name.mkQName(classIdent), NamedAst.Type.Var(classTparam.tpe, classTparam.loc))
               val schemeVal = getDefOrSigScheme(tparams, tpe, uenv0, tenv, List(tconstr), List(classTparam.tpe))
+              val expVal = traverse(exp0)(visitExp(_, env0, uenv0, tenv))
```

Comment:
So this returns a list of zero or one elements? I guess it makes sense. 

In Flix, I think the Traversable type class would ultimately be more flexible and give you back a Validation of an Option.

Reply:
This is possible in Scala too, using implicits. But I can't quite figure out the right way to do it.